### PR TITLE
[PLAT-659] Metamask sign in/up on ACDC

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description

Fix metamask sign in

https://github.com/AudiusProject/audius-client/assets/2731362/4054c538-cc22-4067-84e3-ca2b9525bb74

Requires: https://github.com/AudiusProject/audius-protocol/pull/5398

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

Touches sign up

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

Local w/ linked libs vs. staging
1. Metamask sign up
2. Metamask sign in
3. Metamask reload page
4. Metamask repost
5. Normal sign up

Documentation

Screenshot 2023-06-09 at 5 38 24 PM

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-plat-659

**Review by isaacsolo — APPROVED** [review]

**Review by dylanjeffers — APPROVED** [review]
Awesome and thanks for context. I guess an issue with sdk versions should apply to all packages?

**Comment by raymondjacobson** [comment]
> Awesome and thanks for context. I guess an issue with sdk versions should apply to all packages?

Oh yeah. I can change this across the board. We don't really need it to work in mobile b/c metamask isn't directly supported there, but sounds good.

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

File: packages/mobile/ios/Podfile.lock [hunk 1]
```diff
@@ -1202,4 +1202,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: f1e56ba6cee31ca0cf79cc470496cc0906977d8f
 
-COCOAPODS: 1.12.0
```
Comment (dylanjeffers) [inline packages/mobile/ios/Podfile.lock:1205]: Prob shouldn't change?

**Review by dylanjeffers — APPROVED** [review]
Nice work. Technically package locks could be cleaned up so it's not bopping between sha1 and sha512, but can just have issue till node upgrade

File: packages/mobile/ios/Podfile.lock (hunk 1 above)
Comment (raymondjacobson) [reply packages/mobile/ios/Podfile.lock:1205]: I think this is the latest version? Perhaps everyone should upgrade? installing cocoapods gives you this on our current version of xcode/ruby

Comment (raymondjacobson) [reply packages/mobile/ios/Podfile.lock:1205]: https://github.com/CocoaPods/CocoaPods/releases

Comment (dylanjeffers) [reply packages/mobile/ios/Podfile.lock:1205]: Yeah could be?

Comment (dylanjeffers) [reply packages/mobile/ios/Podfile.lock:1205]: Can just send it and we can all update

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

File: packages/mobile/ios/Podfile.lock (hunk 1 above)
Comment (raymondjacobson) [reply packages/mobile/ios/Podfile.lock:1205]:
```
sudo gem install cocoapods
```
https://guides.cocoapods.org/using/getting-started.html#updating-cocoapods

Comment (raymondjacobson) [reply packages/mobile/ios/Podfile.lock:1205]: I can post out a note

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-plat-659

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-plat-659

File: packages/mobile/src/services/libs.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
 import type { Nullable } from '@audius/common'
-import type { AudiusLibs } from '@audius/sdk/dist/native-libs'
+import type { AudiusLibs } from '@audius/sdk'
 
 // TODO: declare this at the root and use actual audiusLibs type
 declare global {
   interface Window {
-    audiusLibs: any
+    audiusLibs: AudiusLibs
```
Comment (rickyrombo) [inline packages/mobile/src/services/libs.ts:9]: 🥲 thank you

Comment (raymondjacobson) [reply packages/mobile/src/services/libs.ts:9]: believe me i wasn't planning to do this work... it just happened 🌵

**Pull request merged**